backend/settings.py

```diff
@@ -221,7 +223,13 @@
 CANVAS_OAUTH_CLIENT_ID = os.getenv('CANVAS_OAUTH_CLIENT_ID', 'canvas_app_explorer')
 CANVAS_OAUTH_CLIENT_SECRET = os.getenv('CANVAS_OAUTH_CLIENT_SECRET', 'canvas_app_explorer')
 CANVAS_OAUTH_CANVAS_DOMAIN = os.getenv('CANVAS_OAUTH_CANVAS_DOMAIN', 'canvas.instructure.com')
-CANVAS_OAUTH_SCOPES = os.getenv('CANVAS_OAUTH_SCOPES', '').split(',')
+
+# Scopes environment variable provides a way to recover if Canvas changes scope identifiers.
```

Comment:
How can I verify the OAuth access token is created with proper scope?

Reply:
Great! I followed those steps and verified.

1. log message as `[INFO] [oauth.py:73] Redirecting user to https://canvas-test.it.umich.edu/login/oauth2/auth?client_id=17700000000000271&redirect_uri=https%3A%2F%2F4ac5-141-213-175-192.ngrok.io%2Foauth%2Foauth-callback&response_type=code&scope=url%3AGET%7C%2Fapi%2Fv1%2Fcourses%2F%3Acourse_id%2Ftabs+url%3APUT%7C%2Fapi%2Fv1%2Fcourses%2F%3Acourse_id%2Ftabs%2F%3Atab_id&state=7tHHBmfJyMKI`
2. I removed the url:GET|/api/v1/courses/:id, removed the user oauth token in CAE admin page, and re-authenticated myself to CAE. Now, I got this error message "Error occurred! Status: 401; Message: An error occurred while communciating with Canvas. "Insufficient scopes on access token."

I will add those test steps to the original issue, and close this comment.

Reply:
I added a couple more test cases, @zqian, to the comment above.